Improve docs clarity on loading custom checkers (#7276)
PyCQA/pylint: doc/development_guide/how_tos/custom_checkers.rst

.. Note::

    ``my_plugin`` refers to a module called ``my_plugin.py``.
    The preferred way of making this plugin available to pylint is
    by installing it as a package. This can be done either from a packaging index like

>     ``PyPI`` or by installing it from a local source such as with ``pip install .``.

Reviewer: ```suggestion
    ``PyPI`` or by installing it from a local source such as with ``pip install``.
```

>     ``PyPI`` or by installing it from a local source such as with ``pip install .``.

Reviewer: I'm assuming someone that is creating a pylint plugin will have heard of pypi / pip so let's not 😄

    Alternatively, the plugin module can be made available to pylint by
    putting this module's parent directory in your ``PYTHONPATH``
    environment variable.

    If your pylint config has an init-hook that modifies
    ``sys.path`` to include the module's parent directory, this
    will also work, but only if either:

    * the ``init-hook`` and the ``load-plugins`` list are both

>       defined in a pylintrc file, or...

Reviewer 2: ```suggestion
      defined in a configuration file, or...
```

I'm slowly nudging people towards using `pyproject.toml` as the `toml` standard is a bit nicer to work with than the `ini` files.

>     * the ``init-hook`` is passed as a command-line argument and

Reviewer 2: Let's explicitly state what doesn't work

.. Note::

    ``my_plugin`` refers to a module called ``my_plugin.py``.
    The preferred way of making this plugin available to pylint is

>     by installing it as a package, such as with ``pip install .``.

Reviewer 2: ```suggestion
    by installing it as a package. This can be done either from a packaging index like
    ``PyPI`` or by installing it from a local source such as with ``pip install .``.
```

>     Alternatively, this module can be made available to pylint by

Reviewer 2: ```suggestion
    Alternatively, the plugin module can be made available to pylint by
```

    putting this module's parent directory in your ``PYTHONPATH``

>     environment variable or by adding the ``my_plugin.py``
>     file to the ``pylint/checkers`` directory if running from source.

Reviewer 2: ```suggestion
    environment variable.
```

Let's no longer recommend to hack into the `pylint` source. I only foresee additional issues. 😅

Author: Yes, I was a little surprised to see that in there! It's worth noting that I, as so many engineers do, had gone to stack overflow to see what I might be missing. I found one other person complaining about something similar, and they solved it but putting their module in the checkers director 😱 

I left an answer there, linking #7264. I've been updating it as my understanding grows, so hopefully anyone else finding that question will realise that putting things in the pylint source is "not considered best practice" 😄 

https://stackoverflow.com/questions/71590677/cant-load-custom-pylint-module/73250367#73250367

    environment variable or by adding the ``my_plugin.py``
    file to the ``pylint/checkers`` directory if running from source.

    If your pylint config has an init-hook that modifies
    ``sys.path`` to include the module's parent directory, this

>     will also work, but only if:

Reviewer 2: ```suggestion
    will also work, but only if either:
```

Author: Just saw this myself, mortified, good spot thanks @DanielNoord!
